pkg/ebpf/telemetry/errors_telemetry.go

```diff
@@ -50,58 +68,101 @@ type EBPFTelemetry struct {
-var errorsTelemetry *EBPFTelemetry
+// Lock is part of the Locker interface implementation.
+func (e *EBPFTelemetry) Lock() {
+	e.mtx.Lock()
+}
 
-// newEBPFTelemetry initializes a new EBPFTelemetry object
-func newEBPFTelemetry() *EBPFTelemetry {
-	errorsTelemetry = &EBPFTelemetry{
-		mapKeys:   make(map[string]uint64),
-		probeKeys: make(map[string]uint64),
-	}
-	return errorsTelemetry
+// Unlock is part of the Locker interface implementation.
+func (e *EBPFTelemetry) Unlock() {
+	e.mtx.Unlock()
 }
 
-func (b *EBPFTelemetry) setupMapEditors(opts *manager.Options) {
-	if (b.mapErrMap != nil) || (b.helperErrMap != nil) {
+func (e *EBPFTelemetry) setup(opts *manager.Options) {
+	if (e.mapErrMap != nil) || (e.helperErrMap != nil) {
 		if opts.MapEditors == nil {
 			opts.MapEditors = make(map[string]*ebpf.Map)
 		}
 	}
 	// if the maps have already been loaded, setup editors to point to them
-	if b.mapErrMap != nil {
-		opts.MapEditors[probes.MapErrTelemetryMap] = b.mapErrMap.Map()
+	if e.mapErrMap != nil {
+		opts.MapEditors[probes.MapErrTelemetryMap] = e.mapErrMap.Map()
 	}
-	if b.helperErrMap != nil {
-		opts.MapEditors[probes.HelperErrTelemetryMap] = b.helperErrMap.Map()
+	if e.helperErrMap != nil {
+		opts.MapEditors[probes.HelperErrTelemetryMap] = e.helperErrMap.Map()
 	}
 }
 
 // populateMapsWithKeys initializes the maps for holding telemetry info.
 // It must be called after the manager is initialized
-func (b *EBPFTelemetry) populateMapsWithKeys(m *manager.Manager) error {
-	b.mtx.Lock()
-	defer b.mtx.Unlock()
+func (e *EBPFTelemetry) fill(m *manager.Manager) error {
+	e.mtx.Lock()
+	defer e.mtx.Unlock()
 
 	// first manager to call will populate the maps
-	if b.mapErrMap == nil {
-		b.mapErrMap, _ = maps.GetMap[uint64, MapErrTelemetry](m, probes.MapErrTelemetryMap)
+	if e.mapErrMap == nil {
+		e.mapErrMap, _ = maps.GetMap[uint64, MapErrTelemetry](m, probes.MapErrTelemetryMap)
 	}
-	if b.helperErrMap == nil {
-		b.helperErrMap, _ = maps.GetMap[uint64, HelperErrTelemetry](m, probes.HelperErrTelemetryMap)
+	if e.helperErrMap == nil {
+		e.helperErrMap, _ = maps.GetMap[uint64, HelperErrTelemetry](m, probes.HelperErrTelemetryMap)
 	}
 
-	if err := b.initializeMapErrTelemetryMap(m.Maps); err != nil {
+	if err := e.initializeMapErrTelemetryMap(m.Maps); err != nil {
 		return err
 	}
-	if err := b.initializeHelperErrTelemetryMap(); err != nil {
+	if err := e.initializeHelperErrTelemetryMap(); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (b *EBPFTelemetry) initializeMapErrTelemetryMap(maps []*manager.Map) error {
-	if b.mapErrMap == nil {
+func (e *EBPFTelemetry) setProbe(name string, hash uint64) {
+	e.probeKeys[name] = hash
+}
+
+func (e *EBPFTelemetry) isInitialized() bool {
+	return e.mapErrMap != nil && e.helperErrMap != nil
+}
+
+func (e *EBPFTelemetry) forEachMapEntry(yield func(index telemetryIndex, val MapErrTelemetry) bool) {
+	var mval MapErrTelemetry
```

Comment:
I don't think `MapErrTelemetry` or `HelperErrTelemetry` need to be exported, right?

Reply:
i can try and address that in a separate PR (it will require to change the code_gen utility)

Reply:
I think it would just be in `pkg/ebpf/telemetry/types.go`